src/brogue/Monsters.c

```diff
@@ -2807,33 +2807,46 @@ enum directions scentDirection(creature *monst) {
 
 // returns true if the resurrection was successful.
 boolean resurrectAlly(const short x, const short y) {
-    boolean success;
-    creature *monst = firstCreature(&purgatory);
-    if (monst) {
+    creatureIterator allyIterator = iterateCreatures(&purgatory);
+
+    // Prefer most empowered ally.  In case of tie, prefer ally with greatest monsterID (thus
+    // preferring allies found deeper in the dungeon over ones found higher up and preferring
+    // legendary allies over everyone else).
+    creature *monToCheck, *monToRaise = NULL;
+    while (monToCheck = nextCreature(&allyIterator)) {
```

Comment:
As I understand it, @nstoddard's death tweaks mean this won't work any more, since nextCreature will skip over purgatory creatures, since they are dead. I added a commit to unset it during list clean-up. Ideally this needs to be tested

Reply:
It looks like this is necessary and should fix the problem. It might be cleaner to move the line `monst->bookkeepingFlags &= ~(MB_IS_DYING | MB_ADMINISTRATIVE_DEATH | MB_HAS_DIED | MB_IS_FALLING);` here and remove it from `resurrectAlly`, to avoid clearing flags in two places.